File: src/main/java/com/github/morningzeng/toolset/component/ActionBar.java
package com.github.morningzeng.toolset.component;

import com.intellij.openapi.actionSystem.ActionManager;
import com.intellij.openapi.actionSystem.ActionPlaces;
import com.intellij.openapi.actionSystem.ActionToolbar;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.DefaultActionGroup;
import com.intellij.openapi.application.ApplicationManager;
import com.intellij.ui.JBColor;
import com.intellij.ui.components.JBPanel;
import com.intellij.ui.components.JBPanelWithEmptyText;
import com.intellij.util.ui.JBUI.Borders;

import javax.swing.BoxLayout;
import javax.swing.SwingUtilities;
import java.awt.Dimension;

/**
 * @author Morning Zeng
 * @since 2024-06-16
 */
public class ActionBar extends JBPanel<JBPanelWithEmptyText> {

    private final AnAction[] actions;

    public ActionBar(final AnAction... actions) {
        this(true, actions);
    }

    public ActionBar(final boolean horizontal, final AnAction... actions) {
        this.setMaximumSize(new Dimension(Integer.MAX_VALUE, 20));
        this.setBorder(Borders.customLineBottom(JBColor.GRAY));
        this.setLayout(new BoxLayout(this, BoxLayout.LINE_AXIS));

        this.actions = actions;
        ApplicationManager.getApplication().invokeAndWait(() -> {
            final ActionToolbar toolbar = ActionManager.getInstance().createActionToolbar(ActionPlaces.POPUP, new DefaultActionGroup(actions), horizontal);
            toolbar.setTargetComponent(this);
            SwingUtilities.invokeLater(() -> this.add(toolbar.getComponent()));
        });
    }

}

File: src/main/kotlin/com/github/morningzeng/toolbox/ui/bar/ActionBar.kt
package com.github.morningzeng.toolbox.ui.bar

import com.intellij.openapi.actionSystem.ActionManager
import com.intellij.openapi.actionSystem.ActionPlaces
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.DefaultActionGroup
import com.intellij.openapi.application.ApplicationManager
import com.intellij.ui.JBColor
import com.intellij.ui.components.JBPanel
import com.intellij.ui.components.JBPanelWithEmptyText
import com.intellij.util.ui.JBUI
import java.awt.Dimension
import javax.swing.BoxLayout
import javax.swing.SwingUtilities

/**
 * @author Morning Zeng
 * @since 2025-05-20
 */
class ActionBar(
    vararg val actions: AnAction,
    val horizontal: Boolean = true,
) : JBPanel<JBPanelWithEmptyText>() {

    init {
        if (horizontal) {
            maximumSize = Dimension(Int.MAX_VALUE, 20)
            border = JBUI.Borders.customLineBottom(JBColor.GRAY)
        } else {
            Dimension(30, preferredSize.height).apply {
                preferredSize = this
                minimumSize = this
            }
        }
        layout = BoxLayout(this, BoxLayout.LINE_AXIS)
        ApplicationManager.getApplication().invokeAndWait {
            ActionManager.getInstance().createActionToolbar(
                ActionPlaces.POPUP,
                DefaultActionGroup(*actions),
                horizontal
            ).also {
                it.targetComponent = this
                SwingUtilities.invokeLater { add(it.component) }
            }
        }
    }

    override fun setSize(width: Int, height: Int) {
        Dimension(width, height).apply {
            maximumSize = this
            minimumSize = this
            preferredSize = this
        }
    }
}